main.go

```diff
@@ -664,17 +666,34 @@ func transformCompile(args []string) ([]string, error) {
 	}
 
 	if len(privateNameMap) > 0 {
-		outputDirectory := filepath.Dir(flagValue(flags, "-o"))
+		objPath := flagValue(flags, "-o")
+		importcfg := flagValue(flags, "-importcfg")
+		deferred = append(deferred, func() error {
+			importCfg, err := goobj2.ParseImportCfg(importcfg)
+			if err != nil {
+				return err
+			}
 
-		file, err := os.Create(filepath.Join(outputDirectory, garbleMapFile))
-		if err != nil {
-			return nil, err
-		}
-		defer file.Close()
+			pkg, err := goobj2.Parse(objPath, pkgPath, importCfg)
+			if err != nil {
+				return err
+			}
 
-		if err := json.NewEncoder(file).Encode(privateNameMap); err != nil {
-			return nil, err
-		}
+			data, err := json.Marshal(privateNameMap)
+			if err != nil {
+				return err
+			}
+
+			pkg.ArchiveMembers = append(pkg.ArchiveMembers, goobj2.ArchiveMember{
```

Comment:
document here that adding an extra archive header is harmless, and shouldn't break other tools like the linker since our header name is unique